lib/stats/stats-cluster-single.c

```diff
@@ -83,9 +87,7 @@ stats_cluster_single_key_set_with_name(StatsClusterKey *key, guint16 component,
 {
   stats_cluster_key_set(key, component, id, instance, (StatsCounterGroupInit)
   {
-    tag_names, _counter_group_init_with_name, _group_init_equals
+    .counter.name = name, .init = _counter_group_init_with_name, .equals = _group_init_equals
```

Comment:
Nice, I've just found this leak too.

Now that `stats_cluster_single_key_set_with_name()` does not copy the name of the counter, corresponding `StatsClusterKey`s may contain stack addresses. The name is then copied when the actual `StatsCounterGroup` is created.

I think that's good, because we use key setters locally, they are also on the stack, and we register/unregister counters right after that call.